mock/mock.go

```diff
@@ -209,6 +209,12 @@ func (m *Mock) On(methodName string, arguments ...interface{}) *Call {
 	return c
 }
 
+// IsCallExpected checks if a method call is expected.
+func (m *Mock) IsCallExpected(method string, arguments ...interface{}) bool {
+	found, _ := m.findExpectedCall(method, arguments...)
+	return found >= 0
+}
```

Comment:
In our use case, user could optionally setup mock for some of the registered methods (with their expected arguments). When our test suite runs users's tested methods (which calls registered methods), it will have to decide if it should call a mock (if there is one) or it should call the real registered method if there is no mock.